vars/createGithubIssue.groovy

```diff
@@ -6,77 +6,34 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
-def call(Map args = [:]){
-    def failureMessages = args.message
-    List<String> failedComponents = []
-
-    if (failureMessages.size() == 1 && failureMessages[0] == "Build failed") {
-        println("No component failed, skip creating github issue.")
-    }
-    else {
-        for(message in failureMessages.unique()){
-            java.util.regex.Matcher match = (message =~ /(?<=\bcomponent\s).*/)
-            String matched = match[0]
-            println(matched.split(" ")[0].trim())
-            failedComponents.add(matched.split(" ")[0].trim())
-        }
-        /* Due to an existing issue with queryWorkbench plugin breaking OSD during bootstrapping, there are false positive
-           issues getting created against OSD repo. Adding a temp check to ignore issue creation against OSD repo in-case
-           there are more than 1 failures reported for OSD build.
-         */
-        if (failedComponents.contains("OpenSearch-Dashboards") && failedComponents.size() > 1) {
-            failedComponents.removeElement("OpenSearch-Dashboards")
-        }
-
-        def yamlFile = readYaml(file: "manifests/${INPUT_MANIFEST}")
-        def currentVersion = yamlFile.build.version
-
-        for(component in yamlFile.components){
-            if (failedComponents.contains(component.name)) {
-                println("Component ${component.name} failed, creating github issue")
-                compIndex = failedComponents.indexOf(component.name)
-                create_issue(component.name, component.repository, currentVersion, failureMessages[compIndex])
-                sleep(time:3,unit:"SECONDS")
-            }
-        }
-    }
-}
-
-def create_issue(component, url, currentVersion, failedMessage){
-    def versionLabel = "v${currentVersion}"
-    def label = "autocut"
-
-    def message = """***Received Error***: **${failedMessage}**.
-                      The distribution build for ${component} has failed.
-                      Please see build log at ${BUILD_URL}consoleFull""".stripIndent()
-
 
+/** Library to create GitHub issue across opensearch-project repositories.
+ @param Map args = [:] args A map of the following parameters
+ @param args.githubUrl <required> - GitHub repository URL to create issue
+ @param args.issueTitle <required> - GitHub issue title
+ @param args.issueBody <required> - GitHub issue body
+ @param args.label <optional> - GitHub issue label to be attached along with 'untriaged'. Defaults to autocut.
+ */
+void call(Map args = [:]) {
+    label = args.label ?: 'autocut'
```

Comment:
Does this mean if the user passes label argument then `autocut` label won't be applied?

Reply:
Yes